scripts/gen-version.js

```diff
@@ -0,0 +1,11 @@
+const fs = require("fs").promises
+
+fs.readFile("package.json", { encoding: "utf-8" }).then(pkg => {
+    const pkgJson = JSON.parse(pkg)
+    const version = pkgJson.version
+    const versionModule = `export const VERSION = ${JSON.stringify(version, null, 4)}` + "\n"
```

Comment:
Is there a specific reason the space (ie. `4`) is passed as a param to `JSON.stringify`? Shouldn't this be `0`?

Reply:
Ideally, it shouldn't ever make a difference whether it's 4 or 0. It would only apply if the stringified thing would be an array or object - and the version key in `package.json` should only ever be a string.

But *in the absurd case it is something else* it'll lead to nicely-formatted code in `version.ts`? :smile:

Reply:
True 😄 For some reason I was thinking this could turn out in in `"    version"`, but indeed, it would probably be `    "version"`